src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -144,15 +144,16 @@ IECore::ConstCompoundObjectPtr EvaluateLightLinks::computeAttributes( const Scen
 	CompoundObjectPtr result = new CompoundObject;
 	result->members() = inputAttributes->members();
 
+	const PathMatcher &allLights = inPlug()->set( "__lights" )->readable();
```

Comment:
Dangling reference - you need to hold onto the `ConstPathMatcherDataPtr` to keep it alive for as long as you'll use the contents :

```
ConstPathMatcherDataPtr lightsSetData = inPlug()->set( "__lights" );
const PathMatcher &lightsSet = lightsSetData->readable();
```

Reply:
Arg sorry, I was a muppet. I squashed it into what's now 5bdd4dc.